main.go

```diff
@@ -591,7 +593,8 @@ func tryRemoteGoroot(rev string) (
 	if err != nil {
 		return nil, nil, "", "", err
 	}
-	repo, _, err = vs.Repository("git", u)
+	u.User = url.UserPassword(*httpUserFlag, *httpPasswordFlag)
+	repo, _, err = vs.Repository("git", u, *httpPasswordFlag)
```

Comment:
`https://go.googlesource.com/go` is known not to need authentication, so this shouldn't be done here.